tests/modules/images/webp-uploads/webp-uploads-test.php

```diff
@@ -836,4 +998,130 @@ function () {
+	}
+
+	/**
+	 * Return null when looking for a target that is already used
+	 *
+	 * @test
+	 */
+	public function it_should_return_null_when_looking_for_a_target_that_is_already_used() {
+		$attachment_id = $this->factory->attachment->create_upload_object( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/leafs.jpg' );
+		$editor        = new WP_Image_Edit( $attachment_id );
+		$editor->rotate_right()->save();
+
+		$this->assertTrue( $editor->success() );
+		$this->assertNull( webp_uploads_get_next_full_size_key_from_backup( $attachment_id ) );
+	}
+
+	/**
+	 * USe the next available hash for the full size image on multiple image edits
+	 *
+	 * @test
+	 */
+	public function it_should_u_se_the_next_available_hash_for_the_full_size_image_on_multiple_image_edits() {
+		$attachment_id = $this->factory->attachment->create_upload_object( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/leafs.jpg' );
+		$editor        = new WP_Image_Edit( $attachment_id );
+		$editor->rotate_right()->save();
+
+		$this->assertTrue( $editor->success() );
+		$this->assertNull( webp_uploads_get_next_full_size_key_from_backup( $attachment_id ) );
+		// Remove the filter to prevent the usage of the next target.
+		remove_filter( 'wp_update_attachment_metadata', 'webp_uploads_update_attachment_metadata' );
+
+		$editor->rotate_right()->save();
+		$this->assertRegExp( '/full-\d{13}/', webp_uploads_get_next_full_size_key_from_backup( $attachment_id ) );
+	}
+
+	/**
+	 * Save populate the backup sources with the next target
+	 *
+	 * @test
+	 */
+	public function it_should_save_populate_the_backup_sources_with_the_next_target() {
+		// Remove the filter to prevent the usage of the next target.
+		remove_filter( 'wp_update_attachment_metadata', 'webp_uploads_update_attachment_metadata' );
+
+		$attachment_id = $this->factory->attachment->create_upload_object( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/leafs.jpg' );
+		$editor        = new WP_Image_Edit( $attachment_id );
+		$editor->rotate_right()->save();
+		$this->assertTrue( $editor->success() );
+
+		$sources = array( 'image/webp' => 'leafs.webp' );
+		webp_uploads_backup_full_image_sources( $attachment_id, $sources );
+
+		$this->assertSame( array( 'full-orig' => $sources ), get_post_meta( $attachment_id, '_wp_attachment_backup_sources', true ) );
+	}
+
+	/**
+	 * Store the metadata on the next available hash
+	 *
+	 * @test
+	 */
+	public function it_should_store_the_metadata_on_the_next_available_hash() {
+		$attachment_id = $this->factory->attachment->create_upload_object( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/leafs.jpg' );
+		$editor        = new WP_Image_Edit( $attachment_id );
+		$editor->rotate_right()->save();
+		$this->assertTrue( $editor->success() );
+		// Remove the filter to prevent the usage of the next target.
+		remove_filter( 'wp_update_attachment_metadata', 'webp_uploads_update_attachment_metadata' );
+		$editor->rotate_right()->save();
+		$this->assertTrue( $editor->success() );
```

Comment:
Same here:
```suggestion
		$attachment_id = $this->factory->attachment->create_upload_object( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/leafs.jpg' );

		$editor = new WP_Image_Edit( $attachment_id );
		$editor->rotate_right()->save();
		$this->assertTrue( $editor->success() );

		// Remove the filter to prevent the usage of the next target.
		remove_filter( 'wp_update_attachment_metadata', 'webp_uploads_update_attachment_metadata' );
		$editor->rotate_right()->save();
		$this->assertTrue( $editor->success() );
```